dev/pastetools/getclipboard.html

```diff
@@ -0,0 +1,309 @@
+		title: 'Paste plugins',
+		plugins: pastePlugins,
+		selectAll: true
+	} );
+	generateUI( {
+		container: 'basic',
+		title: 'Basic plugins',
+		plugins: basicPlugins,
+		selectAll: true
+	} );
+	generateUI( {
+		container: 'image',
+		title: 'Image plugins',
+		plugins: imagePlugins,
+		selectAll: false
+	} );
+	generateUI( {
+		container: 'other',
+		title: 'Other plugins',
+		plugins: otherPlugins,
+		selectAll: true
+	} );
+
+	updateEditor();
+
+	CKEDITOR.document.getById( 'update' ).on( 'click', updateEditor );
+
+	var rtfButton = document.getElementById( 'rtfData' ),
+		htmlButton = document.getElementById( 'htmlData' );
+
+	rtfButton.onclick = save( document.getElementById( 'rawRtf' ) );
+	htmlButton.onclick = save( document.getElementById( 'rawHtml' ) );
+
+	function save( input ) {
+		return function() {
+			var textBlob = new Blob( [ input.value ], { type: 'text/plain' } );
+			var saveLink = document.createElement( 'a' );
+
+			saveLink.download = input.dataset.name;
+			saveLink.innerHTML = 'Save file';
+			if ( CKEDITOR.env.webkit ) {
+				saveLink.href = window.URL.createObjectURL( textBlob );
+			} else {
+				saveLink.href = window.URL.createObjectURL( textBlob );
+				saveLink.onclick = function( evt ) {
+					document.body.removeChild( evt.target )
+				};
+				saveLink.style.display = 'none';
+				document.body.appendChild( saveLink );
+			}
+			saveLink.click();
+		}
+	}
+
+	function generateUI( options ) {
+		var container = CKEDITOR.document.getById( options.container ),
+			checkboxes = generateOptions( options.plugins, options.selectAll ),
+			title = CKEDITOR.dom.element.createFromHtml( '<p>' + options.title + '</p>' );
+
+		container.append( title );
+		container.append( checkboxes );
+
+		function generateOptions( options, selectAll ) {
+			const optionsContainer = new CKEDITOR.dom.element( 'div' );
+
+			CKEDITOR.tools.array.forEach( options, function( option, i ) {
+				var checkbox = CKEDITOR.dom.element.createFromHtml( '<label><input type="checkbox" class="plugin" value="' + option + '"' + ( selectAll || i === 0 ? ' checked' : '' ) + '> ' + option + '</label>' );
+
+				optionsContainer.append( checkbox );
+			} );
+
+			return optionsContainer;
+		}
+	}
+	var currentInstance;
+
+	function updateEditor() {
+		var plugins = CKEDITOR.document.find( '.plugin:checked' ).toArray(),
+			editorContainer = CKEDITOR.document.getById( 'editor' ),
+			oldInstance = currentInstance;
+
+		plugins = CKEDITOR.tools.array.map( plugins, function( plugin ) {
+			return plugin.getValue();
+		} );
+
+		if ( oldInstance ) {
+			oldInstance.once( 'destroy', function() {
+				setTimeout( createEditor, 500 );
+			} );
+
+			return oldInstance.destroy( true );
+		}
+
+		createEditor();
+
+		function createEditor() {
+			var textarea = new CKEDITOR.dom.element( 'textarea' ),
+				allowedContent = CKEDITOR.document.getById( 'allowedContent' ),
+				extraAllowedContent = CKEDITOR.document.getById( 'extraAllowedContent' ),
```

Comment:
Maybe it makes also sense to add `disallowedContent` option, so we can prefilter HTML a bit for cleaner comparison (where some attributes or styles add noise we would like to skip)?